packages/authority/src/external-id-attestor/external-id-attestor-client.ts

```diff
@@ -8,6 +8,20 @@ type ExternalIdAttestationRequested = PickByType<
   "ExternalIdAttestationRequested"
 >;
 
+/**
+ * ```mermaid
+ * sequenceDiagram
+ *  participant H as Holochain
+ *  actor A as Authority
+ *  participant T as TokenAssessor
+ *  H->>A: Signal (ExternalIdAttestationRequested)
+ *  A->>T: exchangeAccessToken (Username, Password)
```

Comment:
The authority swaps a `codeVerifier` (i.e. the user evidence that they started the web2 authorisation flow) and a `code` (i.e. the authorisation code issued by the web2 identity provider, which acts as secret and reference for the ongoing web2 authorisation flow) for the access token. The user's password never goes near the authority.

We've basically repurposed elements from this flow:
https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-pkce